File: web/src/main/java/no/nav/sbl/dialogarena/rest/Logg.java
package no.nav.sbl.dialogarena.rest;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import static java.lang.String.format;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Logg {

    private String level;
    private String message;
    private String jsFileUrl;
    private String lineNumber;
    private String columnNumber;
    private String url;
    private String userAgent;

    public String getLevel() {
        return level;
    }

    public void setLevel(String level) {
        this.level = level;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getJsFileUrl() {
        return jsFileUrl;
    }

    public void setJsFileUrl(String jsFileUrl) {
        this.jsFileUrl = jsFileUrl;
    }

    public String getLineNumber() {
        return lineNumber;
    }

    public void setLineNumber(String lineNumber) {
        this.lineNumber = lineNumber;
    }

    public String getColumnNumber() {
        return columnNumber;
    }

    public void setColumnNumber(String columnNumber) {
        this.columnNumber = columnNumber;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getUserAgent() {
        return userAgent;
    }

    public void setUserAgent(String userAgent) {
        this.userAgent = userAgent;
    }

    public String melding() {
        return format("%s at %s", message, url);
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/api/informasjon/dto/Logg.kt
package no.nav.sosialhjelp.soknad.api.informasjon.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

enum class LoggLevel {
    ERROR,
    WARN,
    INFO,
}

@JsonIgnoreProperties(ignoreUnknown = true)
data class Logg(
    val level: LoggLevel,
    val message: String?,
    val jsFileUrl: String?,
    val lineNumber: String?,
    val columnNumber: String?,
    val url: String?,
    val userAgent: String?,
) {
    fun melding(): String {
        var useragentWithoutSpaceAndComma = ""
        if (userAgent != null) {
            val useragentWithoutSpace = userAgent.replace(" ", "_")
            useragentWithoutSpaceAndComma = useragentWithoutSpace.replace(",", "_")
        }
        return String.format(
            "jsmessagehash=%s, fileUrl=%s:%s:%s, url=%s, userAgent=%s, melding: %s",
            message.hashCode(),
            jsFileUrl,
            lineNumber,
            columnNumber,
            url,
            useragentWithoutSpaceAndComma,
            message,
        )
    }
}
